custom_connectors/basic_auth/stripe_connector.rb

```diff
@@ -0,0 +1,920 @@
+          }
+        ]
+      end,
+
+      execute: lambda do |_connection, input|
+        get("/v1/subscriptions/#{input['id']}")
+      end,
+
+      output_fields: lambda do |object_definitions|
+        object_definitions["subscription"]
+      end,
+
+      sample_output: lambda do |_connection|
+        get("/v1/subscriptions?limit=1")["data"].first || {}
+      end
+    },
+
+    search_subscriptions: {
+      description: "Search <span class='provider'>subscriptions</span> in " \
+        "<span class='provider'>Stripe</span>",
+
+      input_fields: lambda do
+        [
+          {
+            name: "customer",
+            label: "Customer ID",
+            optional: true,
+            sticky: true
+          },
+          {
+            name: "starting_after",
+            label: "Starting After",
+            hint: "If you make a search request and receive 100 objects, you " \
+              "can specify the ID of last charge here to fetch the next page " \
+              "of the list",
+            optional: true
+          },
+          {
+            name: "ending_before",
+            label: "Ending Before",
+            hint: "If you make a search request and receive 100 objects, you " \
+              "can specify the ID of last charge here to fetch the previous " \
+              "page of the list",
+            optional: true
+          },
+          {
+            name: "status",
+            label: "Status",
+            hint: "The status of the subscriptions to retrieve. One of: " \
+              "trialing, active, past_due, unpaid, canceled, or all.",
+            type: "string",
+            control_type: "select",
+            pick_list: "statuses",
+            optional: true,
+            toggle_hint: "Select from list",
+            toggle_field: {
+              name: "status",
+              label: "Status",
+              type: "string",
+              control_type: "text",
+              optional: true,
+              toggle_hint: "Use custom value",
+              hint: "The status of the subscriptions to retrieve. One of: " \
+                "trialing, active, past_due, unpaid, canceled, or all."
+            },
+            sticky: true
+          }
+        ]
+      end,
+
+      execute: lambda do |_connection, input|
+        param = input.reject { |k, v| v.blank? || k == "status" }
+        if param.length > 0
+          get("/v1/subscriptions",
+              limit: 100,
+              status: (input["status"] || "all")).
+            params(param)
+        else
+          get("/v1/subscriptions",
+              limit: 100,
+              status: (input["status"] || "all"))
+        end
+      end,
+
+      output_fields: lambda do |object_definitions|
+        [
+          { name: "data", type: "array", of: "object",
+            properties: object_definitions["subscription"] }
+        ]
+      end,
+
+      sample_output: lambda do |_connection|
+        get("/v1/subscriptions?limit=1") || {}
```

Comment:
If response is blank, should substitute with an empty array instead of empty hash.